client/lib/upgrades/actions/free-checkout.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import { cartItems, emptyTemporaryCart, fillInAllCartItemAttributes } from 'lib/cart-values';
+import productsListFactory from 'lib/products-list';
+import { fullCreditsPayment } from 'lib/store-transactions';
+import { submitTransaction } from './checkout';
+
+const productsList = productsListFactory();
+
+export function freeTrial( siteID, plan, onComplete ) {
```

Comment:
@Tug Good idea to move this out of `actions/checkout.js`. What do you think about naming it `actions/free-trials.js`, and naming this method `startFreeTrial`? My rationale for these changes:

- Right now, this is all free-trial-specific. Some "free checkouts" (purchases made with credits) still go through checkout.
- We generally include a verb with actions.